src/otx/core/data/dataset/base.py

```diff
@@ -158,11 +179,27 @@ def _get_img_data_and_shape(self, img: Image) -> tuple[np.ndarray, tuple[int, in
             msg = "Cannot get image data"
             raise RuntimeError(msg)
 
-        img_data = self._cache_img(key=key, img_data=img_data.astype(np.uint8))
+        if roi:
+            # extract ROI from image
+            shape = roi["shape"]
+            h, w = img_data.shape[:2]
+            x1, y1, x2, y2 = (
+                int(np.clip(np.trunc(shape["x1"] * w), 0, w)),
+                int(np.clip(np.trunc(shape["y1"] * h), 0, h)),
+                int(np.clip(np.ceil(shape["x2"] * w), 0, w)),
+                int(np.clip(np.ceil(shape["y2"] * h), 0, h)),
+            )
+            if x1 >= x2 or y1 >= y2:
```

Comment:
@kprokofi Could you also add this? Thanks

```suggestion
if (x2 - x1) * (y2 - y1) <= 0:
    msg = "ROI has zero or negative area."
    raise ValueError(msg)
```

Reply:
I think it is the same condition, but I rewrote it.

Reply:
https://github.com/openvinotoolkit/training_extensions/pull/4105/commits/5563ba9038fe56e5e5633857410fbce4d51de800